app/docs/md/conventions/components.md

```diff
@@ -0,0 +1,253 @@
+
+<doc-code filename="app/components/my-card.mjs">
+
+```javascript
+import CustomElement from '@enhance/custom-element'
+
+export default class MyCard extends CustomElement {
+  connectedCallback() {
+    this.heading = this.querySelector('h5')
+  }
+
+  render({ html, state }) {
+    const { attrs={} } = state
+    const { title='default' } = attrs
+    return html`
+      <style>
+        :host {
+            position: relative;
+            display: flex;
+            flex-direction: column;
+            min-width: 0;
+            word-wrap: break-word;
+            color: black;
+            background-color: #fff;
+            background-clip: border-box;
+            border: 1px solid rgba(0,0,0,.125);
+            border-radius: 0.25rem;
+        }
+        .card-img {
+            width: 100%;
+            border-top-left-radius: calc(0.25rem - 1px);
+            border-top-right-radius: calc(0.25rem - 1px);
+        }
+        .card-body {
+            flex: 1 1 auto;
+            padding: 1.25rem;
+        }
+        .card-title {
+            margin-bottom: 0.75rem;
+            font-size: 1.25rem;
+            font-weight: 500;
+        }
+      </style>
+      <slot name="image"></slot>
+      <div class="card-body font-sans">
+        <h5 class="card-title">${title}</h5>
+        <slot></slot>
+      </div>
+    `
+  }
+
+  static get observedAttributes() {
+    return [ 'title' ]
+  }
+
+  titleChanged(value) {
+    this.heading.textContent = value
+  }
+}
+
+customElements.define('my-card', MyCard)
+```
+
+</doc-code>
+
+You may be thinking that the `render` function looks familiar and you would be right. These `render` functions are Enhance Elements. This enables us to share rendering logic between the client side and server side so any Enhance Component will be server side renderable.
+
+When an Enhance component is server-side rendered it is "enhanced" with an attribute to indicate that the slotting algorithm and style transform have already been run. The attribute look like this:
+
+```html
+<my-card enhanced=”✨”></my-card>
+```
+
+The client-side code will look for this attribute and only run if your component hasn’t already been "enhanced". Thus avoiding an unnecessary render pass.
+
+If you are have an existing Enhance Element you can always import it into your Component and use it as your `render` function.
+
+<doc-callout level="none" mark="✨">
+
+**[Learn about the HTML render function](/docs/elements/html)**
+
+</doc-callout>
+
+## Rehydration
+Enhance Components handle rehydration by listening to attribute changes. Any change to an attribute listed in the `observedAttributes` will trigger a `<attribute name>Changed` method. For example if you are observing the `title` attribute of our `my-card` component any time that attribute value is updated the `titleChanged` method will be executed. This enables you to write surgical DOM updates which will always be the most performant way to update your page.
```

Comment:
~~handle rehydration~~ Updates to Enhance Components are triggered by attribute changes.